source/adapters/level_zero/usm.cpp

```diff
@@ -753,50 +724,53 @@ ur_usm_pool_handle_t_::ur_usm_pool_handle_t_(ur_context_handle_t Context,
     pNext = const_cast<void *>(BaseDesc->pNext);
   }
 
-  auto MemProvider =
-      umf::memoryProviderMakeUnique<L0HostMemoryProvider>(Context, nullptr)
-          .second;
-
-  HostMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Host])
-          .second;
-
-  for (auto device : Context->Devices) {
-    MemProvider =
-        umf::memoryProviderMakeUnique<L0DeviceMemoryProvider>(Context, device)
-            .second;
-    DeviceMemPools.emplace(
-        std::piecewise_construct, std::make_tuple(device),
-        std::make_tuple(umf::poolMakeUnique<usm::DisjointPool, 1>(
-                            {std::move(MemProvider)},
-                            this->DisjointPoolConfigs
-                                .Configs[usm::DisjointPoolMemType::Device])
-                            .second));
-
-    MemProvider =
-        umf::memoryProviderMakeUnique<L0SharedMemoryProvider>(Context, device)
-            .second;
-    SharedMemPools.emplace(
-        std::piecewise_construct, std::make_tuple(device),
-        std::make_tuple(umf::poolMakeUnique<usm::DisjointPool, 1>(
-                            {std::move(MemProvider)},
-                            this->DisjointPoolConfigs
-                                .Configs[usm::DisjointPoolMemType::Shared])
-                            .second));
-
-    MemProvider = umf::memoryProviderMakeUnique<L0SharedReadOnlyMemoryProvider>(
-                      Context, device)
-                      .second;
-    SharedReadOnlyMemPools.emplace(
-        std::piecewise_construct, std::make_tuple(device),
-        std::make_tuple(
-            umf::poolMakeUnique<usm::DisjointPool, 1>(
-                {std::move(MemProvider)},
-                this->DisjointPoolConfigs
-                    .Configs[usm::DisjointPoolMemType::SharedReadOnly])
-                .second));
+  ur_result_t Ret;
+  std::tie(Ret, PoolManager) =
+      usm::pool_manager<usm::pool_descriptor>::create();
+  if (Ret) {
+    urPrint("urUSMPoolCreate: unexpected internal error\n");
```

Comment:
Use logger when introducing new log messages.

Reply:
Alright, just make sure that if the #1032 is merged first, the logger is used instead.